src/main/kotlin/app/revanced/patches/tiktok/misc/loginfallback/patch/TikTokLoginFallbackPatch.kt

```diff
@@ -0,0 +1,41 @@
+package app.revanced.patches.tiktok.misc.loginfallback.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.impl.BytecodeData
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patches.tiktok.misc.loginfallback.annotations.TikTokWebLoginCompatibility
+
+@Patch
+@Name("tiktok-web-login")
+@Description("Login with google use web auth.")
+@TikTokWebLoginCompatibility
+@Version("0.0.1")
+class TikTokLoginFallbackPatch : BytecodePatch() {
+    override fun execute(data: BytecodeData): PatchResult {
+        val googleOneTapAuth = data.findClass("Lcom/bytedance/lobby/google/GoogleOneTapAuth;")!!.resolve()
+        val googleOneTapAuthCheck = googleOneTapAuth.methods.first { it.returnType == "Z" && it.parameters.size == 0 }
```

Comment:
Z could change eventually. Turn this into a fingerprint so it can be found regardless of the method name.

Reply:
I am confusing. The target method is the unique with return type boolean and 0 paramaters in GoogleOneTapAuth class. I don't use method name here.

Reply:
@oSumAtrIX